TicTacToe/TicTacToe.js

```diff
@@ -0,0 +1,157 @@
+class Game {
+    moves = 0;
+    user = 'X';
+    board = [['-', '-', '-'], ['-', '-', '-'], ['-', '-', '-']];
+    messages = "Game - START!";
+    complete = false;
+
+    reset() {
+        this.moves = 0;
+        this.user = 'X';
+        this.board = [['-', '-', '-'], ['-', '-', '-'], ['-', '-', '-']];
+        this.messages = "Game - START!";
+        this.complete = false;
+    }
+
+    getMoves() {
+        return this.moves;
+    };
+
+    displayBoard() {
+        let text = '';
+        for (let i = 0; i < 3; i++) {
+            for (let k = 0; k < 3; k++) {
+                text += " " + this.board[i][k];
+            }
+            text += "\n";
+        }
+        console.log(text);
+        return text;
+    };
+
+    getMessages() {
+        return this.messages;
+    }
+
+    makeMove(x, y) {
+        this.messages = "";
+        if (this.complete) {
+            this.messages = "GAME COMPLETED";
+            return;
+        }
+
+        if (this.board[x][y] != '-') {
+            this.messages = "Invalid move! Try again!";
+            if (this.user == 'X') {
+                return 'O';
+            } else {
+                return 'X';
+            }
+        }
+
+        this.board[x][y] = this.user;
+        this.moves++;
+
+        if (this.isWin(x, y)) {
+            this.messages = this.user + " is the winner!";
```

Comment:
In this case, it's not too necessary, but you can look into string interpolation. Useful for formatting strings with arbitrary data embedded in it. This case would look like:
```javascript
this.messages = `${this.user} is the winner!`;
```